src/js/content.js

```diff
@@ -2,24 +2,68 @@ chrome.runtime.onMessage.addListener((request) => {
   let walkthrough_data = request.walkthrough_data;
   console.log(walkthrough_data);
 
-  const box = createWalkthroughContainer(walkthrough_data);
-
+  const box = createWalkthroughContainer(walkthrough_data.google);
   document.querySelector("body").appendChild(box);
   console.log(box);
+
+  const updatePopup = function(title, content) {
+    let heading = document.querySelector(".wt-title")
+    let body = document.querySelector(".wt-body");
+
+
+    heading.innerText = title;
+    body.innerText = content;
+  }
+
+
+  const forwardBtn = document.querySelector(".forward")
+  let currentStep = 0;
+  let contentStep = 0;
+  let data = walkthrough_data.google.steps[currentStep];
+  forwardBtn.addEventListener("click", function() {
+  
+    // check if we have any tutorial steps
+    if(!data) {   
+      console.log("End of tutorial");
+      updatePopup("End of tutorial", "Have a nice day!")
+    } else {
+      
+      // if only 1 substep exists, display the data and move on to the next tutorial step on button click
+      // otherwise, traverse through substeps on button click
+      if(data.content.length === 1) {
+        updatePopup(data.title, data.content[0].text);
+        contentStep = 0;
+        currentStep++;
+      } else if(data.content.length > 1 && data.content[contentStep]) {  
+        updatePopup(data.title, data.content[contentStep].text);
+        contentStep++;
+      }
+
+      // if at the end of the substeps, on button click move on to the next tutorial step.
+      if(!data.content[contentStep]) {
+        contentStep = 0;
+        currentStep++;
+      }
+
```

Comment:
Couldn't `contentStep++` be included outside of the `if/else if` (i.e include it in line 47 instead of repeating three times)?

Reply:
I believe if we move `contentStep++` to line 47, we run into an infinite loop sort of issue. At some point, the extension will flip back and forth between 2 steps when we click the forward button.

Reply:
Oh, that makes sense. My fault, I read the variable names wrong.